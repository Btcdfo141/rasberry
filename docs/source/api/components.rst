:mod:`spencerassistant.components`
===============================

air\_quality
--------------------------------------------

.. automodule:: spencerassistant.components.air_quality
    :members:
    :undoc-members:
    :show-inheritance:

alarm\_control\_panel
--------------------------------------------

.. automodule:: spencerassistant.components.alarm_control_panel
    :members:
    :undoc-members:
    :show-inheritance:

binary\_sensor
--------------------------------------------

.. automodule:: spencerassistant.components.binary_sensor
    :members:
    :undoc-members:
    :show-inheritance:

camera
---------------------------

.. automodule:: spencerassistant.components.camera
    :members:
    :undoc-members:
    :show-inheritance:

calendar
---------------------------

.. automodule:: spencerassistant.components.calendar
    :members:
    :undoc-members:
    :show-inheritance:

climate
---------------------------

.. automodule:: spencerassistant.components.climate
    :members:
    :undoc-members:
    :show-inheritance:

conversation
---------------------------

.. automodule:: spencerassistant.components.conversation
    :members:
    :undoc-members:
    :show-inheritance:

cover
---------------------------

.. automodule:: spencerassistant.components.cover
    :members:
    :undoc-members:
    :show-inheritance:

device\_tracker
---------------------------

.. automodule:: spencerassistant.components.device_tracker
    :members:
    :undoc-members:
    :show-inheritance:

fan
---------------------------

.. automodule:: spencerassistant.components.fan
    :members:
    :undoc-members:
    :show-inheritance:

light
---------------------------

.. automodule:: spencerassistant.components.light
    :members:
    :undoc-members:
    :show-inheritance:

lock
---------------------------

.. automodule:: spencerassistant.components.lock
    :members:
    :undoc-members:
    :show-inheritance:

media\_player
---------------------------

.. automodule:: spencerassistant.components.media_player
    :members:
    :undoc-members:
    :show-inheritance:

notify
---------------------------

.. automodule:: spencerassistant.components.notify
    :members:
    :undoc-members:
    :show-inheritance:

remote
---------------------------

.. automodule:: spencerassistant.components.remote
    :members:
    :undoc-members:
    :show-inheritance:

switch
---------------------------

.. automodule:: spencerassistant.components.switch
    :members:
    :undoc-members:
    :show-inheritance:

sensor
-------------------------------------

.. automodule:: spencerassistant.components.sensor
    :members:
    :undoc-members:
    :show-inheritance:

vacuum
-------------------------------------

.. automodule:: spencerassistant.components.vacuum
    :members:
    :undoc-members:
    :show-inheritance:

water\_heater
-------------------------------------

.. automodule:: spencerassistant.components.water_heater
    :members:
    :undoc-members:
    :show-inheritance:

weather
---------------------------

.. automodule:: spencerassistant.components.weather
    :members:
    :undoc-members:
    :show-inheritance:

webhook
---------------------------

.. automodule:: spencerassistant.components.webhook
    :members:
    :undoc-members:
    :show-inheritance:
