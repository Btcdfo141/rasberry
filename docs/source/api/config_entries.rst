.. _config_entries_module:

:mod:`spencerassistant.config_entries`
-----------------------------------

.. automodule:: spencerassistant.config_entries
    :members:
