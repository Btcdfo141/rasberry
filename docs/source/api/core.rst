.. _core_module:

:mod:`spencerassistant.core`
-------------------------

.. automodule:: spencerassistant.core
    :members: