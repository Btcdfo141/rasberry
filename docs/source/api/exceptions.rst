.. _exceptions_module:

:mod:`spencerassistant.exceptions`
-------------------------------

.. automodule:: spencerassistant.exceptions
    :members:
