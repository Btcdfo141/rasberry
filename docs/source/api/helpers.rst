:mod:`spencerassistant.helpers`
============================

.. automodule:: spencerassistant.helpers
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.aiohttp\_client
-------------------------------------

.. automodule:: spencerassistant.helpers.aiohttp_client
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.area\_registry
------------------------------------

.. automodule:: spencerassistant.helpers.area_registry
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.check\_config
-----------------------------------

.. automodule:: spencerassistant.helpers.check_config
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.collection
--------------------------------

.. automodule:: spencerassistant.helpers.collection
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.condition
-------------------------------

.. automodule:: spencerassistant.helpers.condition
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.config\_entry\_flow
-----------------------------------------

.. automodule:: spencerassistant.helpers.config_entry_flow
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.config\_entry\_oauth2\_flow
-------------------------------------------------

.. automodule:: spencerassistant.helpers.config_entry_oauth2_flow
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.config\_validation
----------------------------------------

.. automodule:: spencerassistant.helpers.config_validation
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.data\_entry\_flow
---------------------------------------

.. automodule:: spencerassistant.helpers.data_entry_flow
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.debounce
------------------------------

.. automodule:: spencerassistant.helpers.debounce
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.deprecation
---------------------------------

.. automodule:: spencerassistant.helpers.deprecation
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.device\_registry
--------------------------------------

.. automodule:: spencerassistant.helpers.device_registry
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.discovery
-------------------------------

.. automodule:: spencerassistant.helpers.discovery
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.dispatcher
--------------------------------

.. automodule:: spencerassistant.helpers.dispatcher
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity
----------------------------

.. automodule:: spencerassistant.helpers.entity
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_component
---------------------------------------

.. automodule:: spencerassistant.helpers.entity_component
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_platform
--------------------------------------

.. automodule:: spencerassistant.helpers.entity_platform
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_registry
--------------------------------------

.. automodule:: spencerassistant.helpers.entity_registry
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_values
------------------------------------

.. automodule:: spencerassistant.helpers.entity_values
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entityfilter
----------------------------------

.. automodule:: spencerassistant.helpers.entityfilter
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.event
---------------------------

.. automodule:: spencerassistant.helpers.event
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.icon
--------------------------

.. automodule:: spencerassistant.helpers.icon
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.integration\_platform
-------------------------------------------

.. automodule:: spencerassistant.helpers.integration_platform
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.intent
----------------------------

.. automodule:: spencerassistant.helpers.intent
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.json
--------------------------

.. automodule:: spencerassistant.helpers.json
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.location
------------------------------

.. automodule:: spencerassistant.helpers.location
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.logging
-----------------------------

.. automodule:: spencerassistant.helpers.logging
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.network
-----------------------------

.. automodule:: spencerassistant.helpers.network
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.restore\_state
------------------------------------

.. automodule:: spencerassistant.helpers.restore_state
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.script
----------------------------

.. automodule:: spencerassistant.helpers.script
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.service
-----------------------------

.. automodule:: spencerassistant.helpers.service
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.signal
-----------------------------

.. automodule:: spencerassistant.helpers.signal
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.state
---------------------------

.. automodule:: spencerassistant.helpers.state
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.storage
-----------------------------

.. automodule:: spencerassistant.helpers.storage
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.sun
-------------------------

.. automodule:: spencerassistant.helpers.sun
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.system\_info
----------------------------------

.. automodule:: spencerassistant.helpers.system_info
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.temperature
---------------------------------

.. automodule:: spencerassistant.helpers.temperature
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.template
------------------------------

.. automodule:: spencerassistant.helpers.template
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.translation
---------------------------------

.. automodule:: spencerassistant.helpers.translation
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.typing
----------------------------

.. automodule:: spencerassistant.helpers.typing
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.update\_coordinator
-----------------------------------------

.. automodule:: spencerassistant.helpers.update_coordinator
   :members:
   :undoc-members:
   :show-inheritance:
