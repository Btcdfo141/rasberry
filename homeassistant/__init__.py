"""Init file for Home Assistant."""


test2
