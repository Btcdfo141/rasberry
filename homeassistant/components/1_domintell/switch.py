import logging

from homeassistant.components.switch import SwitchEntity

# Create a logger
_LOGGER = logging.getLogger(__name__)


async def async_setup_platform(hass, config, add_entities, discovery_info=None):
    """Set up the Domintell switch platform."""
    # This function is executed when Home Assistant loads your switch platform.
    # This is where you'd normally process any discovered devices.
    # For now, we're just going to create a single virtual switch entity.

    if discovery_info is None:
        return

    switch_name = discovery_info
    _LOGGER.info(f"Setting up Domintell virtual switch: {switch_name}")

    switch_entity = DomintellVirtualSwitch(switch_name)
    add_entities([switch_entity])


class DomintellVirtualSwitch(SwitchEntity):
    """Representation of a Domintell virtual switch."""

    def __init__(self, name):
        """Initialize the switch."""
        self._name = name
        self._unique_id = f"domintell_{name.lower().replace(' ', '_')}"
        self._state = False

    @property
    def unique_id(self) -> str:
        """Return a unique ID."""
        _LOGGER.info(f"Unique ID: {self._unique_id}")
        return self._unique_id

    @property
    def name(self):
        """Return the name of the switch."""
        return self._name

    @property
    def is_on(self):
        """Return True if the switch is on."""
        return self._state

    async def async_turn_on(self, **kwargs):
        """Turn on the switch."""
        # In a real implementation, you'd have logic here to turn on the actual device.
        _LOGGER.info("Domintell switch turned ON")
        self._state = True
        self.async_write_ha_state()

    async def async_turn_off(self, **kwargs):
        """Turn off the switch."""
        # In a real implementation, you'd have logic here to turn off the actual device.
        _LOGGER.info("Domintell switch turned  OFF")
        self._state = False
        self.async_write_ha_state()
