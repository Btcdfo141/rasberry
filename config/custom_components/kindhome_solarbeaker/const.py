from datetime import timedelta

from homeassistant.const import Platform

DOMAIN="kindhome_solarbeaker"
PLATFORMS = [Platform.COVER]

UPDATE_INTERVAL = timedelta(seconds=20)

SERVICE_UUID = "75c276c3-8f97-20bc-a143-b354244886d4"
TITLE="Kindhome Solarbeaker"

DATA_DEVICE = "device"
DATA_COOR = "coordinator"
