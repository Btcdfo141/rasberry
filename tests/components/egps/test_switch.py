"""Test the switch functionality."""

from collections.abc import Callable, Generator
from unittest.mock import MagicMock, patch

from pyegps.exceptions import EgpsException
import pytest
from syrupy.assertion import SnapshotAssertion

from homeassistant.components.egps.const import DOMAIN
from homeassistant.components.homeassistant import (
    DOMAIN as HOME_ASSISTANT_DOMAIN,
    SERVICE_UPDATE_ENTITY,
)
from homeassistant.components.switch import (
    DOMAIN as SWITCH_DOMAIN,
    SERVICE_TURN_OFF,
    SERVICE_TURN_ON,
)
from homeassistant.config_entries import ConfigEntryState
from homeassistant.const import STATE_OFF, STATE_ON, Platform
from homeassistant.core import HomeAssistant
from homeassistant.exceptions import HomeAssistantError
from homeassistant.helpers import device_registry as dr, entity_registry as er

from tests.common import MockConfigEntry


@pytest.fixture(name="mock_get_device")
def patch_get_device(pyegps_device_mock: MagicMock) -> Generator[MagicMock, None, None]:
    """Fixture to patch the `get_device` api method."""
    with patch(
        "homeassistant.components.egps.get_device", return_value=pyegps_device_mock
    ) as mock:
        yield mock


async def test_switch_setup(
    hass: HomeAssistant,
    device_registry: dr.DeviceRegistry,
    entity_registry: er.EntityRegistry,
    valid_config_entry: MockConfigEntry,
    mock_get_device: MagicMock,
    snapshot: SnapshotAssertion,
) -> None:
    """Test a successful setup of device switches."""

    entry = valid_config_entry
    entry.add_to_hass(hass)

    await hass.config_entries.async_setup(entry.entry_id)
    await hass.async_block_till_done()

    assert entry.state == ConfigEntryState.LOADED
    assert entry.entry_id in hass.data[DOMAIN]

    device_entry = device_registry.async_get_device(
        identifiers={(DOMAIN, entry.unique_id)}
    )
    entries = er.async_entries_for_device(entity_registry, device_entry.id)
    switches = [entry for entry in entries if entry.domain == Platform.SWITCH]
    assert len(switches) == 4

    switch = switches[0]
    assert switch == snapshot

    assert await hass.config_entries.async_unload(entry.entry_id)
    await hass.async_block_till_done()


async def _test_switch_on_off(
    hass: HomeAssistant, entity_id: str, dev: MagicMock
) -> None:
    """Call switch on/off service."""
    await hass.services.async_call(
        SWITCH_DOMAIN,
        SERVICE_TURN_ON,
        {"entity_id": entity_id},
        blocking=True,
    )
    assert hass.states.get(entity_id).state == STATE_ON

    await hass.services.async_call(
        SWITCH_DOMAIN,
        SERVICE_TURN_OFF,
        {"entity_id": entity_id},
        blocking=True,
    )

    assert hass.states.get(entity_id).state == STATE_OFF


async def _test_switch_on_exeception(
    hass: HomeAssistant, entity_id: str, dev: MagicMock
) -> None:
    """Call switch on service with USBError side effect."""
    dev.switch_on.side_effect = EgpsException
    with pytest.raises(HomeAssistantError):
        await hass.services.async_call(
            HOME_ASSISTANT_DOMAIN,
            SERVICE_TURN_ON,
            {"entity_id": entity_id},
            blocking=True,
        )

    dev.switch_on.side_effect = None


async def _test_switch_off_exeception(
    hass: HomeAssistant, entity_id: str, dev: MagicMock
) -> None:
    """Call switch off service with USBError side effect."""
    dev.switch_off.side_effect = EgpsException
    with pytest.raises(HomeAssistantError):
        await hass.services.async_call(
            SWITCH_DOMAIN,
            SERVICE_TURN_OFF,
            {"entity_id": entity_id},
            blocking=True,
        )
    dev.switch_off.side_effect = None


async def _test_switch_update_exception(
    hass: HomeAssistant, entity_id: str, dev: MagicMock
) -> None:
    """Call switch off service with USBError side effect."""
    dev.get_status.side_effect = EgpsException
    with pytest.raises(HomeAssistantError):
        await hass.services.async_call(
            SWITCH_DOMAIN,
            SERVICE_UPDATE_ENTITY,
            {"entity_id": entity_id},
            blocking=True,
        )
    dev.get_status.side_effect = None


@pytest.mark.parametrize(
    "test_function",
    [
        _test_switch_on_off,
        _test_switch_on_exeception,
        _test_switch_off_exeception,
        _test_switch_update_exception,
    ],
)
async def test_switch_services(
    hass: HomeAssistant,
    device_registry: dr.DeviceRegistry,
    entity_registry: er.EntityRegistry,
    valid_config_entry: MockConfigEntry,
    mock_get_device: MagicMock,
    test_function: Callable,
) -> None:
    """Test a successful setup of device switches."""

    entry = valid_config_entry
    entry.add_to_hass(hass)

    await hass.config_entries.async_setup(entry.entry_id)
    await hass.async_block_till_done()

    assert entry.state == ConfigEntryState.LOADED
    assert entry.entry_id in hass.data[DOMAIN]

    device_entry = device_registry.async_get_device(
        identifiers={(DOMAIN, entry.unique_id)}
    )
    entries = er.async_entries_for_device(entity_registry, device_entry.id)
    switches = [entry for entry in entries if entry.domain == Platform.SWITCH]

    switch = switches[0]
    device_mock = mock_get_device.return_value
    await test_function(hass, switch.entity_id, device_mock)

    assert await hass.config_entries.async_unload(entry.entry_id)
    await hass.async_block_till_done()
