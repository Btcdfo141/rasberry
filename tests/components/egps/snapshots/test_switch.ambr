# serializer version: 1
# name: test_switch_setup
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': None,
    'entity_id': 'switch.mockedusbdevice_socket_0',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SwitchDeviceClass.OUTLET: 'outlet'>,
    'original_icon': None,
    'original_name': 'Socket 0',
    'platform': 'egps',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': None,
    'unique_id': 'DYPS:00:11:22_0',
    'unit_of_measurement': None,
  })
# ---
